components/menu/menu-item-link.js

```diff
@@ -57,7 +61,7 @@ class MenuItemLink extends MenuItemMixin(LitElement) {
 	render() {
 		const rel = this.target ? 'noreferrer noopener' : undefined;
 		return html`
-			<a href="${ifDefined(this.href)}" rel="${ifDefined(rel)}" target="${ifDefined(this.target)}" tabindex="-1">
+			<a download="${ifDefined(this.download)}" href="${ifDefined(this.href)}" rel="${ifDefined(rel)}" target="${ifDefined(this.target)}" tabindex="-1">
```

Comment:
Would our `download` attribute be required?  According to the specs:
> Prompts the user to save the linked URL instead of navigating to it. Can be used with or without a value:
>
> Without a value, the browser will suggest a filename/extension, generated from various sources:
The Content-Disposition HTTP header
The final segment in the URL path
The media type (from the (Content-Type header, the start of a data: URL, or Blob.type for a blob: URL)
Defining a value suggests it as the filename. / and \ characters are converted to underscores (_). Filesystems may forbid other characters in filenames, so browsers will adjust the suggested name if necessary.

This implies to me that both of these scenarios are valid:
``` javascript
<a download href="...">

<a download="some name" href="...">
```
But with the current rendering, if the consumer does not provide a value for `download`, the user will not be prompted to save instead of navigating.  Is that true?

Reply:
That's a good question... worth testing, but I think just setting a boolean attribute will still bind to the string (not sure what value it'll end up with, maybe empty?).

Reply:
The `download` attribute would not be required.

The following 3 scenarios are valid:
```
// Navigates. Rendered when no 'download' attribute is passed
<a href="...">

// Downloads, and generates file name in the browser. Rendered when 'download' attribute is passed with no value
<a download href="...">

// Downloads, with custom file name. Rendered when 'download' attribute is passed with value
<a download="custom-name" href="...">
```

`ifDefined()` will render the attribute even if it does not have a value, so the `download` attribute is set without a value in that case.

Reply:
Should I add demo examples and tests with `download` and `download="value"`?

Reply:
> That's a good question... worth testing, but I think just setting a boolean attribute will still bind to the string (not sure what value it'll end up with, maybe empty?).

`<d2l-menu-item-link download>` will gives the MenuItemLink object a `download` value of `''`, empty-string.

Reply:
Thanks, yeah that's what I was wondering.  That might be ok.  I'm not sure if the empty string matters or not.  If it does, it could be avoided using something roughly along these lines:

```javascript
const hasDownloadName = this.download && this.download.length > 0;

return hasDownloadName 
    ? html`<a download="${this.download}" ...></a>` 
    : html`<a ?download="${!!this.download}" ...></a>`; 
```

Reply:
If you write `<a download=""></a>` to the DOM, it will render as `<a download></a>`. Attributes with empty-string values have their values removed.